lib/reverse_sentence.rb

```diff
@@ -1,6 +1,66 @@
 # A method to reverse the words in a sentence, in place.
-# Time complexity: ?
-# Space complexity: ?
+# Time complexity: O(n) where n is the number of characters in the sentence. This is because it is essentially multiplying
+# the number of characters in each word by 1 --> for each word, reverse each character in that word. So, it ends up being (1 * number of characters in word 1) + (1 * number of characters in word 2) etc. for each word,
+# which simplifies to adding together the number of characters in each word, which is essentailly O(n)
+# Space complexity: O(1)
 def reverse_sentence(my_sentence)
-  raise NotImplementedError
+  if my_sentence == nil
+    return my_sentence
+  end
+  
+  length = my_sentence.length
+  (length / 2).times do |i|
+    my_sentence[i], my_sentence[length - 1 - i] = my_sentence[length - 1 - i], my_sentence[i]
+  end
```

Comment:
Notice that you are reversing things twice in this method.  Could you dry things up by making a helper method?